### PR TITLE
🏃Improve unit test coverage for KCP

### PR DESCRIPTION
**Pull request by wfernandes**
**What this PR does / why we need it**:
Just adding some more tests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Ref https://github.com/kubernetes-sigs/cluster-api/issues/2779

### CODE REVIEW

File: controlplane/kubeadm/internal/kubeadm_config_map_test.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ kubernetesVersion: v1.16.1
 	delete(kubeadmConfigNoKey.Data, clusterConfigurationKey)
 
 	kubeadmConfigBadData := kconf.DeepCopy()
-	kubeadmConfigBadData.Data[clusterConfigurationKey] = `foobar`
+	kubeadmConfigBadData.Data[clusterConfigurationKey] = `something`
```
Comment (neolit123) [inline controlplane/kubeadm/internal/kubeadm_config_map_test.go:50]: is this change needed?

File: controlplane/kubeadm/internal/workload_cluster_test.go [hunk 2]
```diff
@@ -474,6 +474,85 @@ kubernetesVersion: v1.16.1
...
+
+	kubeadmConfigBadData := kubeadmConfig.DeepCopy()
+	kubeadmConfigBadData.Data[clusterConfigurationKey] = `foobar`
+
+	g := NewWithT(t)
+	scheme := runtime.NewScheme()
+	g.Expect(corev1.AddToScheme(scheme)).To(Succeed())
+	tests := []struct {
+		name      string
+		imgRepo   string
```
Comment (neolit123) [inline controlplane/kubeadm/internal/workload_cluster_test.go:503]: should be imageRepository to match existing tests in kubeadm_config_map_test.go

File: controlplane/kubeadm/internal/workload_cluster_test.go [hunk 3]
```diff
@@ -474,6 +474,85 @@ kubernetesVersion: v1.16.1
...
+		objs      []runtime.Object
```
Comment (neolit123) [inline controlplane/kubeadm/internal/workload_cluster_test.go:504]: objects

File: controlplane/kubeadm/internal/workload_cluster_test.go [hunk 4]
```diff
@@ -474,6 +474,85 @@ kubernetesVersion: v1.16.1
...
+				return
+			}
+			g.Expect(err).ToNot(HaveOccurred())
+			var actualConfig corev1.ConfigMap
+			g.Expect(w.Client.Get(
+				ctx,
+				ctrlclient.ObjectKey{Name: kubeadmConfigKey, Namespace: metav1.NamespaceSystem},
+				&actualConfig,
+			)).To(Succeed())
+			g.Expect(actualConfig.Data[clusterConfigurationKey]).To(ContainSubstring("imageRepository: myspecialrepo.io"))
```
Comment (neolit123) [inline controlplane/kubeadm/internal/workload_cluster_test.go:551]: should match `tt.imgRepo` here.

File: controlplane/kubeadm/internal/kubeadm_config_map_test.go (hunk 1 above)
Comment (wfernandes) [reply controlplane/kubeadm/internal/kubeadm_config_map_test.go:50]: I was getting a linting error saying `foobar` was used three times and that I should change it to a constant.

File: controlplane/kubeadm/internal/workload_cluster_test.go (hunk 3 above)
Comment (wfernandes) [reply controlplane/kubeadm/internal/workload_cluster_test.go:504]: do you want me to change all the references of `objs` to `objects` in this file.

Comment (neolit123) [reply controlplane/kubeadm/internal/workload_cluster_test.go]: ok, let's leave it objs given the other tests use that.

**Comment by neolit123** [comment]
thanks @wfernandes 
/approve
